app/controllers/integrations/base_controller.rb

```diff
@@ -53,11 +51,10 @@ def release_params
     { commit: commit, author: user }
   end
 
-  def create_new_release
-    unless project.last_release_contains_commit?(commit)
-      release_service = ReleaseService.new(project)
-      release_service.release!(release_params)
-    end
+  def find_or_create_release
+    latest_release = project.releases.order(:id).last # FIXME: ordering by number would be good here
```

Comment:
nvm ... numbers should always be increasing ... so need a new validation for that ... next PR